packages/editor-ui/src/components/layouts/ResourcesListLayout.vue

```diff
@@ -0,0 +1,405 @@
+			}
+
+			return (this.resources as IResource[]).filter((resource) => {
+				if (this.isOwnerSubview && this.$store.getters['settings/isEnterpriseFeatureEnabled'](EnterpriseEditionFeature.Sharing)) {
+					return !!(resource.ownedBy && resource.ownedBy.id === this.currentUser.id);
+				}
+
+				return true;
+			});
+		},
+		filterKeys(): string[] {
+			return Object.keys(this.filters);
+		},
+		filteredAndSortedSubviewResources(): IResource[] {
+			const filtered: IResource[] = this.subviewResources.filter((resource: IResource) => {
+				let matches = true;
+
+				if (this.filters.ownedBy) {
+					matches = matches && !!(resource.ownedBy && resource.ownedBy.id === this.filters.ownedBy);
+				}
+
+				if (this.filters.sharedWith) {
+					matches = matches && !!(resource.sharedWith && resource.sharedWith.find((sharee) => sharee.id === this.filters.sharedWith));
+				}
+
+				if (this.filters.search) {
+					const searchString = this.filters.search.toLowerCase();
+
+					matches = matches && resource.name.toLowerCase().includes(searchString);
+				}
+
+				if (this.additionalFiltersHandler) {
+					matches = this.additionalFiltersHandler(resource, this.filters, matches);
+				}
+
+				return matches;
+			});
+
+			return filtered.sort((a, b) => {
+				switch (this.sortBy) {
+					case 'lastUpdated':
+						return (new Date(b.updatedAt)).valueOf() - (new Date(a.updatedAt)).valueOf();
+					case 'lastCreated':
+						return (new Date(b.createdAt)).valueOf() - (new Date(a.createdAt)).valueOf();
+					case 'nameAsc':
+						return a.name.localeCompare(b.name);
+					case 'nameDesc':
+						return b.name.localeCompare(a.name);
+					default:
+						return 0;
+				}
+			});
+		},
+		resourcesNotOwned(): IResource[] {
+			return (this.resources as IResource[]).filter((resource) => {
+				return resource.ownedBy && resource.ownedBy.id !== this.currentUser.id;
+			});
+		},
+	},
+	methods: {
+		async onMounted() {
+			await this.initialize();
+
+			this.loading = false;
+			this.$nextTick(this.focusSearchInput);
+		},
+		resetFilters() {
+			Object.keys(this.filters).forEach((key) => {
+				this.filters[key] = Array.isArray(this.filters[key]) ? [] : '';
+			});
+
+			this.resettingFilters = true;
+			this.sendFiltersTelemetry('reset');
+		},
+		focusSearchInput() {
+			if (this.$refs.search) {
+				(this.$refs.search as Vue & { focus: () => void }).focus();
+			}
+		},
+		setOwnerSubview(active: boolean) {
+			this.isOwnerSubview = active;
+		},
+		getTelemetrySubview(): string {
+			return this.$locale.baseText(`${this.resourceKey as IResourceKeyType}.menu.${this.isOwnerSubview ? 'my' : 'all'}`);
+		},
+		sendSubviewTelemetry() {
+			this.$telemetry.track('User changed credentials sub view', {
+				sub_view: this.getTelemetrySubview(),
+			});
+		},
+		sendSortingTelemetry() {
+			this.$telemetry.track('User changed sorting in cred list', {
+				sub_view: this.getTelemetrySubview(),
+				sorting: this.sortBy,
+			});
+		},
```

Comment:
hmm seems like these should be refactored out

Reply:
Good catch! Fixed.